sklearn/tree/tree.py

```diff
@@ -414,22 +414,21 @@ def predict(self, X, check_input=True):
         check_is_fitted(self, 'tree_')
         X = self._validate_X_predict(X, check_input)
         proba = self.tree_.predict(X)
-        n_samples = X.shape[0]
 
         # Classification
         if is_classifier(self):
             if self.n_outputs_ == 1:
                 return self.classes_.take(np.argmax(proba, axis=1), axis=0)
 
             else:
-                predictions = np.zeros((n_samples, self.n_outputs_))
+                predictions = []
 
                 for k in range(self.n_outputs_):
-                    predictions[:, k] = self.classes_[k].take(
+                    predictions.append(self.classes_[k].take(
                         np.argmax(proba[:, k], axis=1),
-                        axis=0)
+                        axis=0))
 
-                return predictions
+                return np.array(predictions).T
```

Comment:
this would try to figure the dtype of the array, right? how much is it slower than the status quo?